File: src/Other/MaterialDesign_Styles_Kit/AccentColorConverter.cs
using System;
#if SLMIGRATION
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Media;
using System.Globalization;
#else
using Windows.UI;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media;
#endif

namespace MaterialDesign_Styles_Kit
{
    public class AccentColorConverter : IValueConverter
    {
#if SLMIGRATION
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
#else
        public object Convert(object value, Type targetType, object parameter, string language)
#endif
        {
            if (value is SolidColorBrush)
            {
                Color oldColor = ((SolidColorBrush)value).Color;
                Byte newR = oldColor.R > 40 ? (Byte)(oldColor.R - 40) : (Byte)0;
                Byte newG = oldColor.G > 40 ? (Byte)(oldColor.G - 40) : (Byte)0;
                Byte newB = oldColor.B > 40 ? (Byte)(oldColor.B - 40) : (Byte)0;
                Color newColor = Color.FromArgb(oldColor.A, newR, newG, newB);
                return new SolidColorBrush(newColor);
            }
            return value;
        }

#if SLMIGRATION
        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
#else
        public object ConvertBack(object value, Type targetType, object parameter, string language)
#endif
        {
            throw new NotImplementedException();
        }
    }
}

File: src/Other/MaterialDesign_Styles_Kit/AccentColorConverter.fs
namespace MaterialDesign_Styles_Kit

open System
#if SLMIGRATION
open System.Windows
open System.Windows.Controls
open System.Windows.Data
open System.Windows.Media
open System.Globalization
#else
open Windows.UI
open Windows.UI.Xaml.Data
open Windows.UI.Xaml.Media
#endif

type AccentColorConverter() =
    interface IValueConverter with
        member this.Convert(value, targetType, parameter, language) =
            if value :? SolidColorBrush then
                let oldColor = (value :?> SolidColorBrush).Color
                let newR = if oldColor.R > byte(40) then byte (oldColor.R - byte(40)) else byte 0
                let newG = if oldColor.G > byte(40) then byte (oldColor.G - byte(40)) else byte 0
                let newB = if oldColor.B > byte(40) then byte (oldColor.B - byte(40)) else byte 0
                let newColor = Color.FromArgb(oldColor.A, newR, newG, newB)
                new SolidColorBrush(newColor) :> obj
            else
                value

        member this.ConvertBack(value, targetType, parameter, language) =
            raise (NotImplementedException())
